Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean

```diff
@@ -610,6 +614,124 @@ theorem adjust_f {w₁ : InfinitePlace K} (B : ℝ≥0) (hf : ∀ w, w ≠ w₁
 
 end convexBodyLT
 
+section convexBodyLT'
+
+open  Metric ENNReal NNReal
+
+open Classical
+
+variable (f : InfinitePlace K → ℝ≥0) (w₀ : {w : InfinitePlace K // IsComplex w})
+
+/-- A version of `convexBodyLT` with an additional condition at a fixed complex place. This is
+needed to ensure the element constructed is not real, see for example
+`exists_primitive_element_lt_of_isComplex`.
+-/
+abbrev convexBodyLT' : Set (E K) :=
+  (Set.univ.pi (fun w : { w : InfinitePlace K // IsReal w } ↦ ball 0 (f w))) ×ˢ
+  (Set.univ.pi (fun w : { w : InfinitePlace K // IsComplex w } ↦
+    if w = w₀ then {x | |x.re| < 1 ∧ |x.im| < (f w : ℝ) ^ 2} else ball 0 (f w)))
+
+theorem convexBodyLT'_mem {x : K} :
+    mixedEmbedding K x ∈ (convexBodyLT' K f w₀) ↔
+      (∀ w : InfinitePlace K, w ≠ w₀ → w x < f w) ∧
+      |(w₀.val.embedding x).re| < 1 ∧ |(w₀.val.embedding x).im| < (f w₀: ℝ) ^ 2 := by
+  simp_rw [mixedEmbedding, RingHom.prod_apply, Set.mem_prod, Set.mem_pi, Set.mem_univ,
+    forall_true_left, Pi.ringHom_apply, apply_ite, mem_ball_zero_iff, ← Complex.norm_real,
+    embedding_of_isReal_apply, norm_embedding_eq, Subtype.forall, Set.mem_setOf_eq]
+  refine ⟨fun ⟨h₁, h₂⟩ ↦ ⟨fun w h_ne ↦ ?_, ?_⟩, fun ⟨h₁, h₂⟩ ↦ ⟨fun w hw ↦ ?_, fun w hw ↦ ?_⟩⟩
+  · by_cases hw : IsReal w
+    · exact norm_embedding_eq w _ ▸ h₁ w hw
+    · specialize h₂ w (not_isReal_iff_isComplex.mp hw)
+      rwa [if_neg (by exact Subtype.ne_of_val_ne h_ne)] at h₂
+  · simpa [if_true] using h₂ w₀.val w₀.prop
+  · exact h₁ w (ne_of_isReal_isComplex hw w₀.prop)
+  · by_cases h_ne : w = w₀
+    · simpa [h_ne]
+    · rw [if_neg (by exact Subtype.ne_of_val_ne h_ne)]
+      exact h₁ w h_ne
+
+theorem convexBodyLT'_symmetric (x : E K) (hx : x ∈ (convexBodyLT' K f w₀)) :
```

Comment:
We usually call this results `neg_mem` or something similar (since `symmetric` does not appear in the statement).

Reply:
Good point. I changed it everywhere in the file.